tests/Unit/CallPowerTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Tests\Unit;
+
+use Exception;
+use Tests\TestCase;
+use Illuminate\Support\Facades\Bus;
+use Chompy\Jobs\CreateCallPowerPostInRogue;
+
+class CallPowerTest extends TestCase
+{
+    /**
+     * Test that a post with a completed status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testCompletedCallStatus()
+    {
+        // Fake the job being dispatched.
+        Bus::fake();
+
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'completed',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertSuccessful();
+
+        Bus::assertDispatched(CreateCallPowerPostInRogue::class);
+    }
+
+    /**
+     * Test that a post with a busy status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testBusyCallStatus()
+    {
+        // Fake the job being dispatched.
+        Bus::fake();
+
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'busy',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertSuccessful();
+
+        Bus::assertDispatched(CreateCallPowerPostInRogue::class);
+    }
+
+    /**
+     * Test that a post with an invalid CallPower campaign id is not made.
+     *
+     * @return void
+     */
+    public function testFailedPostWithInvalidCallPowerCampaignId()
+    {
+        // If the provided Action ID doesn't exist in Rogue, this method throws:
+        $this->rogueMock->shouldReceive('getActionIdFromCallPowerCampaignId')->andThrow(new Exception(500));
```

Comment:
@DFurnes I'm a little stuck on this one because it does throw an exception, as we want it to, but then the whole test errors out: 
```
There was 1 error:

1) Tests\Unit\CallPowerTest::testFailedPostWithInvalidCallPowerCampaignId
Exception: 500
```
How do we get around this to say we want this to happen? I added `$this->expectException(Exception::class);` at the bottom to assert this but the test errors out before it gets to that point I think.

Reply:
I _think_ I figured it out to move the `$this->expectException(Exception::class);` to before the job is called so it doesn't error out?

Reply:
Yes, that sounds right to me! That should be telling PHPUnit to "expect" that type of exception to be thrown, and so it swallows it up (and I _think_ actually fails the test if it _isn't_ thrown). 👍